src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1077,25 +1077,30 @@ private void gather(final ResourceResolver resolver, final List<MapEntry> entrie
      */
     private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final String key, final MapEntry entry) {
 
-        if (entry==null){
+        if (entry == null) {
+            log.trace("trying to add null entry for {}", key);
             return false;
-        }
-
-        List<MapEntry> entries = entryMap.get(key);
-        if (entries == null) {
-            entries = new ArrayList<>();
-            entries.add(entry);
-            // and finally sort list
-            Collections.sort(entries);
-            entryMap.put(key, entries);
         } else {
```

Comment:
When entry is null, the return statement will be executed. So no need to do a "else" here

Reply:
The intent was to have both "returns" at the same level. Gimme a sec.